client/my-sites/checkout/composite-checkout/hooks/use-maybe-jetpack-intro-coupon-code.ts

```diff
@@ -1,48 +1,36 @@
-import { isJetpackProductSlug, isJetpackPlanSlug } from '@automattic/calypso-products';
+import { isJetpackSearch } from '@automattic/calypso-products';
 import { useMemo } from 'react';
-import { useSelector } from 'react-redux';
-import { getJetpackSaleCoupon } from 'calypso/state/marketing/selectors';
 import type { MinimalRequestCartProduct } from '@automattic/shopping-cart';
 
-const JETPACK_INTRO_COUPON_CODE = 'FRESHPACK';
+const JETPACK_SEARCH_INTRO_COUPON_CODE = 'FIRSTYEAR50';
+
+const isYearly = ( product: MinimalRequestCartProduct ) =>
+	! product.product_slug.endsWith( '_monthly' );
+const isNewPurchase = ( product: MinimalRequestCartProduct ) =>
+	product.extra?.purchaseType === 'renewal';
```

Comment:
Is this supposed to be `product.extra?.purchaseType !== 'renewal'`?

Reply:
Great catch 👍